asllib/doc/ASLTypingReference.tex

```diff
@@ -5756,73 +5871,94 @@ \section{TypingRule.LEConcat \label{sec:TypingRule.LEConcat}}
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 \chapter{Typing of Slices \label{chap:typingslices}}
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+\hypertarget{def-annotateslice}{}
+The function
+\[
+  \annotateslice(\overname{\staticenvs}{\tenv} \aslsep \overname{\slice}{\vs})
+  \aslto
+  \overname{\slice}{\vsp} \cup \overname{\TTypeError}{\TypeErrorConfig}
+\]
+annotates a slices $\vs$ in the static environment $\tenv$,
+resulting in an annotated slice $\vsp$.
+A type error is returned, if one is detected.
 
-Annotating a slice~\texttt{slice} in an environment $\tenv$
-(\texttt{annotate\_slice env slice}) results in the slicing of the pair \texttt{(offset,length)} and one of the following applies:
+One of the following applies:
 \begin{itemize}
 \item TypingRule.SliceSingle (see \secref{TypingRule.SliceSingle}),
 \item TypingRule.SliceLength (see \secref{TypingRule.SliceLength}),
 \item TypingRule.SliceRange (see \secref{TypingRule.SliceRange}),
 \item TypingRule.SliceStar (see \secref{TypingRule.SliceStar}).
 \end{itemize}
 
-\section{TypingRule.SliceSingle \label{sec:TypingRule.SliceSingle}}
+\hypertarget{def-annotateslices}{}
+The function
+\[
+  \annotateslices(\overname{\staticenvs}{\tenv} \aslsep \overname{\slice^*}{\slices})
+  \aslto
+  \overname{\slice^*}{\slicesp} \cup \overname{\TTypeError}{\TypeErrorConfig}
+\]
+annotates a list of slices $\slices$ in the static environment $\tenv$,
+resulting in an annotated list of slices $\slicesp$.
+A type error is returned, if one is detected.
 
-  \subsection{Prose}
-   Annotating a slice~\texttt{slice} in an environment $\tenv$
-(\texttt{annotate\_slice env slice}) results in the pair \texttt{(offset, length)} and all of the following apply:
-   \begin{itemize}
-   \item \texttt{slice} gives an index \texttt{i};
-   \item \texttt{(offset, length)} is the result of applying TypingRule.SliceLength to \texttt{i, i+:1}.
-   \end{itemize}
+The relevant rule is given by:
```

Comment:
Why the forward pointer sorry?

Reply:
I like listing all the rules in the chapter but the previous rules go with the "One of the following applies:" and this rule does not.

Reply:
Sorry I am not sure I understand. Let’s discuss Monday?